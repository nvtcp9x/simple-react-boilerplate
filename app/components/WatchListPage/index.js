// @flow

import * as React from 'react';
import { Row, Col } from 'reactstrap'
import { URL_IMG, IMG_SIZE_LARGE } from '../../../utils/constants'
import { StyledImage } from '../../../utils/commonStyle'
import styled from 'styled-components'

const StyledP = styled.p`
  font-size: 1em;
  line-height: 1.4em;
`

const StyledColumn = styled(Col)`
  margin: auto;
`
const StyledRow = styled(Row)`
  border: 1px solid #e3e3e3;
  box-shadow: 0 2px 8px rgba(0,0,0,0.1);
  background-color: #fff;
  margin-bottom: 15px;
`
const StyledTextH1 = styled.h1`
  font-size:1.2rem;
  font-weight: 700;
`

const WatchListPage = ({ bookMarkedMovies }: Array) =>
  (
    bookMarkedMovies.map(bookMarkedMovie =>
      (
        <StyledRow key={bookMarkedMovie.id}>
          <Col xs={6} sm={3} md={3} key={bookMarkedMovie.id}>
            <StyledImage src={URL_IMG + IMG_SIZE_LARGE + bookMarkedMovie.poster_path} className="img-fluid" />
          </Col>
          <StyledColumn xs={6} sm={9} md={9}>
            <StyledTextH1>{bookMarkedMovie.original_title}</StyledTextH1>
            <StyledTextH1>Overview: </StyledTextH1>
            <StyledP>{bookMarkedMovie.length > 70 ? `${bookMarkedMovie.substring(0, 70)}...` : bookMarkedMovie.overview}</StyledP>
            <StyledTextH1>Release_date</StyledTextH1>
            <StyledP>{bookMarkedMovie.release_date}</StyledP>
          </StyledColumn>
        </StyledRow>
      ))
  )

export default WatchListPage
