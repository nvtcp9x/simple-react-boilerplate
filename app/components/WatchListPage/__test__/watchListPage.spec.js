import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import WatchListPage from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('WatchListPage component', () => {
  const mockProps = {
    bookMarkedMovies: [{
      id: 123,
      poster_path: 'http',
      overview: 'overvierw',
    },
    ],
  };

  const component = shallow(<WatchListPage {...mockProps} />);
  it('should match to snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
  it('should return overview with over 70 character', () => {
    component.setProps({
      bookMarkedMovies: [{
        id: 123,
        poster_path: 'http',
        overview: "Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages",
      },
      ],
    });
  })
});
