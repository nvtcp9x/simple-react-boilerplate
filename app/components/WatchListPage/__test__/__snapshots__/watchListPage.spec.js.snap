// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchListPage component should match to snapshot 1`] = `
Array [
  <Styled(Row)
    key="123"
  />,
]
`;
